internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -98,6 +98,17 @@ func (nviu NodeVersionInfoUpdate) NodeUpgradeableCheck(client clientset.Interfac
 	if err != nil {
 		return err
 	}
+	if nviu.Current.IsControlPlane() {
```

Comment:
If you fix the ToleratesClusterVersion above,  you don't need to add this:
it will be covered a little below on kubernetes.AllControlPlanesMatchVersion which calls AllControlPlanesMatchVersionWithVersioningInfo .
That check can be moved up if necessary.

Reply:
Nope, this `if` condition is part of function `NodeUpgradeableCheck` and `ToleratesClusterVersion` or `EqualsClusterVersion` will no check if local `kubelet` package is in correct version to corresponding `kube-apiserver`

Reply:
It have to stay otherwise we will hit problem from https://github.com/SUSE/avant-garde/issues/2028#issuecomment-720449984

Reply:
I am fine with having a check in here.